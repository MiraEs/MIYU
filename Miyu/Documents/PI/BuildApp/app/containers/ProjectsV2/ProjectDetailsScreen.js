import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import { InteractionManager } from 'react-native';
import {
	TabbedPager,
	withScreen,
} from 'BuildLibrary';
import NavigationBarIconButton from '../../components/navigationBar/NavigationBarIconButton';
import helpers from '../../lib/helpers';
import { connect } from 'react-redux';
import TrackingActions from '../../lib/analytics/TrackingActions';
import { withNavigation } from '@expo/ex-navigation';
import TeamTab from './ProjectDetails/TeamTab';
import ShoppingListTab from './ProjectDetails/ShoppingListTab';
import PhotosTab from './ProjectDetails/PhotosTab';
import { getProject } from '../../reducers/helpers/projectsReducerHelper';

export class ProjectDetailsScreen extends Component {

	constructor(props) {
		super(props);
		this.tabs = [{
			name: 'Shopping List',
			component: (
				<ShoppingListTab
					key="shoppingList"
					projectId={props.projectId}
				/>
			),
		}, {
			name: 'Team',
			component: (
				<TeamTab
					key="team"
					projectId={props.projectId}
				/>
			),
		}, {
			name: 'Photos',
			component: (
				<PhotosTab
					key="shoppingList"
					projectId={props.projectId}
				/>
			),
		}];
	}

	componentDidMount() {
		InteractionManager.runAfterInteractions(() => {
			this.props.navigator.updateCurrentRouteParams({
				onSettingsPress: this.onProjectDetailsSettingsPress,
				showSettingsButton: !!this.props.project.isOwnedByUser,
				projectName: this.props.project.name,
			});
		});
	}

	componentWillReceiveProps({ project: { name } = {} }) {
		const {
			project: {
				name: oldName,
			} = {},
		} = this.props;
		if (name && name !== oldName) {
			this.props.navigator.updateCurrentRouteParams({
				projectName: name,
			});
		}
	}

	setScreenTrackingInformation() {
		return {
			name: 'build:app:lists:projects:details',
		};
	}

	onProjectDetailsSettingsPress = () => {
		this.props.navigator.push('projectDetailsSettings', {
			projectId: this.props.projectId,
		});
	};

	render() {
		return (
			<TabbedPager
				tabs={this.tabs}
				initialPage={this.props.selectedTabIndex}
				scrollEnabled={false}
			/>
		);
	}
}

ProjectDetailsScreen.displayName = 'Project Details Screen';

ProjectDetailsScreen.route = {
	navigationBar: {
		visible: true,
		title({ projectName }) {
			return projectName;
		},
		renderRight(route) {
			if (route.params.showSettingsButton) {
				return (
					<NavigationBarIconButton
						onPress={route.params.onSettingsPress}
						iconName={helpers.getIcon('settings')}
						trackAction={TrackingActions.PROJECT_DETAILS_SETTINGS}
					/>
				);
			}
		},
	},
};

ProjectDetailsScreen.propTypes = {
	projectId: PropTypes.number.isRequired,
	loading: PropTypes.bool.isRequired,
	navigator: PropTypes.shape({
		push: PropTypes.func,
		updateCurrentRouteParams: PropTypes.func,
	}),
	project: PropTypes.shape({
		isOwnedByUser: PropTypes.bool,
		name: PropTypes.string,
	}),
	selectedTabIndex: PropTypes.number,
};

ProjectDetailsScreen.defaultProps = {
	selectedTabIndex: 0,
};

const mapStateToProps = (state, ownProps) => {
	return {
		loading: state.projectsReducer.isLoading,
		project: getProject(state.projectsReducer.projects, ownProps.projectId),
	};
};

export default withNavigation(connect(mapStateToProps)(withScreen(ProjectDetailsScreen)));
