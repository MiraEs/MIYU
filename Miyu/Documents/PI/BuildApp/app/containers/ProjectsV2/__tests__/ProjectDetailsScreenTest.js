'use strict';

jest.mock('../../../lib/analytics/tracking');
jest.mock('react-native');
jest.mock('BuildNative');
jest.mock('BuildLibrary');
jest.mock('../../../components/navigationBar/NavigationBarIconButton', () => 'NavigationBarIconButton');
jest.mock('../../../components/Library/Pager/Pager', () => 'Pager');
jest.mock('../../../components/Library/Pager/PagerTabBar', () => 'PagerTabBar');
jest.mock('../ProjectDetails/TeamTab', () => 'TeamTab');
jest.mock('../ProjectDetails/ShoppingListTab', () => 'ShoppingListTab');
jest.mock('../ProjectDetails/PhotosTab', () => 'PhotosTab');

import { ProjectDetailsScreen } from '../ProjectDetailsScreen';
import React from 'react';

const defaultProps = {
	error: '',
	loading: false,
	projectId: 1,
	navigator: {
		push: jest.fn(),
		updateCurrentRouteParams: jest.fn(),
	},
};

describe('ProjectDetailsScreen component', () => {
	it('should render correctly', () => {
		const tree = require('react-test-renderer').create(
			<ProjectDetailsScreen {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render the shopping list tab', () => {
		const tree = require('react-test-renderer').create(
			<ProjectDetailsScreen
				{...defaultProps}
				selectedTabIndex={0}
			/>
		);
		expect(tree).toMatchSnapshot();
	});

	it('should render the team tab', () => {
		const tree = require('react-test-renderer').create(
			<ProjectDetailsScreen
				{...defaultProps}
				selectedTabIndex={1}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render the photos tab', () => {
		const tree = require('react-test-renderer').create(
			<ProjectDetailsScreen
				{...defaultProps}
				selectedTabIndex={2}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

});
