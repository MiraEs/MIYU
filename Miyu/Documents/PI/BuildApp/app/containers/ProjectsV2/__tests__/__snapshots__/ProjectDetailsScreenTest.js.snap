// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetailsScreen component should render correctly 1`] = `
<TabbedPager
  initialPage={0}
  scrollEnabled={false}
  tabs={
    Array [
      Object {
        "component": <ShoppingListTab
          projectId={1}
    />,
        "name": "Shopping List",
      },
      Object {
        "component": <TeamTab
          projectId={1}
    />,
        "name": "Team",
      },
      Object {
        "component": <PhotosTab
          projectId={1}
    />,
        "name": "Photos",
      },
    ]
  }
/>
`;

exports[`ProjectDetailsScreen component should render the photos tab 1`] = `
<TabbedPager
  initialPage={2}
  scrollEnabled={false}
  tabs={
    Array [
      Object {
        "component": <ShoppingListTab
          projectId={1}
    />,
        "name": "Shopping List",
      },
      Object {
        "component": <TeamTab
          projectId={1}
    />,
        "name": "Team",
      },
      Object {
        "component": <PhotosTab
          projectId={1}
    />,
        "name": "Photos",
      },
    ]
  }
/>
`;

exports[`ProjectDetailsScreen component should render the shopping list tab 1`] = `
<TabbedPager
  initialPage={0}
  scrollEnabled={false}
  tabs={
    Array [
      Object {
        "component": <ShoppingListTab
          projectId={1}
    />,
        "name": "Shopping List",
      },
      Object {
        "component": <TeamTab
          projectId={1}
    />,
        "name": "Team",
      },
      Object {
        "component": <PhotosTab
          projectId={1}
    />,
        "name": "Photos",
      },
    ]
  }
/>
`;

exports[`ProjectDetailsScreen component should render the team tab 1`] = `
<TabbedPager
  initialPage={1}
  scrollEnabled={false}
  tabs={
    Array [
      Object {
        "component": <ShoppingListTab
          projectId={1}
    />,
        "name": "Shopping List",
      },
      Object {
        "component": <TeamTab
          projectId={1}
    />,
        "name": "Team",
      },
      Object {
        "component": <PhotosTab
          projectId={1}
    />,
        "name": "Photos",
      },
    ]
  }
/>
`;
